core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -1032,15 +988,8 @@ class DynamicListenerConfig(server: KafkaBroker) extends BrokerReconfigurable wi
       if (oldConfig.effectiveListenerSecurityProtocolMap(listenerName) != newConfig.effectiveListenerSecurityProtocolMap(listenerName))
         throw new ConfigException(s"Security protocol cannot be updated for existing listener $listenerName")
     }
-    if (!newAdvertisedListeners.contains(newConfig.interBrokerListenerName))
+    if (!oldAdvertisedListeners.contains(newConfig.interBrokerListenerName))
```

Comment:
Is `interBrokerListenerName` a kind of dynamic config? If not, `KafkaConfig` has checked it in initialization.